### PR TITLE
feat:  [M3-8326],  [M3-8055] - Create Linode - Create Using Command Line A/B Test

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝
This PR uses the new feature flag for A/B testing for the Command line tools.

## Changes  🔄
List any change relevant to the reviewer.
- Created and used feature flag for A/B testing.
- Added event emitter to track the metrics for the trackable elements in the DX tool Modal.
- Created Metrics in the launch darkly and mapped to A/B testing experiment.

## Target release date 🗓️
9/16

## How to test 🧪
- Checkout the branch and run the app
- Navigate to Create command line tools or View code snippets Modal and click on tabs, copy icon and resources.
- Verify experiments results in LD reflects the clicks. Reach out me for further help.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Comment by jaalah-akamai** [comment]
Metrics not firing for me:
- ❌ Initial opening of modal
- ❌ Linode CLI Tab Selection
- ❌ cURL Tab Selection
- ❌ All Resource Links

Metrics working:
- ✅ Linode CLI Copy Code Snippet Icon
- ✅ cURL Copy Code Snippet Icon
- ✅ Integrations - Ansible - Copy Code Snippet Icon
- ✅ Integrations - Terraform - Copy Code Snippet Icon
- ✅ SDKs - Python - Copy Code Snippet Icon
- ✅ SDKs - Go - Copy Code Snippet Icon

File: packages/manager/src/constants.ts [hunk 1]
```diff
@@ -294,3 +294,34 @@ export const FIREWALL_GET_STARTED_LINK =
...
+  OPEN_MODAL: 'A/B Test: Step 1 : DX Tools Open Modal',
+  SDK_GO_CODE_SNIPPET:
+    'A/B Test: Step 2 : SDK: GO copy code snippet (copy icon)',
+  SDK_GO_RESOURCE_LINKS: 'A/B Test: Step 2 : DX Tools SDK GO resources links',
+  SDK_PYTHON_CODE_SNIPPET:
+    'A/B Test: Step 2 : SDK: Python copy code snippet (copy icon)',
+  SDK_PYTHON_RESOURCE_LINKS:
+    'A/B Test: Step 2 : DX Tools SDK Python resources links',
+  SDK_TAB_SELECTION: 'A/B Test: Step 2 : DX Tools SDK tab selection',
+};
```
Comment (jaalah-akamai) [inline packages/manager/src/constants.ts:327]: For others: These relate to `Event Key` for metrics

**Review by jaalah-akamai — APPROVED** [review]
I think we're gathering metrics for the core of the experiment, which is the button wording `Create Using Command Line` vs `View Code Snippets` where 100% of the audience is being directed to one of those two variations. Every other metric is just nice-to-have.

File: packages/manager/src/constants.ts (hunk 1 above)
Comment (cpathipa) [reply packages/manager/src/constants.ts:327]: Yea, These are the metric keys from LD portal, where we are passing them from code base when firing events for trackable elements.

**Comment by jdamore-linode** [comment]
@cpathipa mind pulling in the latest from `develop` to get the tests passing?

**Comment by cpathipa** [comment]
@jdamore-linode done!

**Comment by jdamore-linode** [comment]
Thanks @cpathipa 🙌

**Comment by cpathipa** [comment]
> Metrics not firing for me:
> 
> * ❌ All Resource Links
> 
> Metrics working:
> 
> * ✅ Initial opening of modal
> * ✅ Linode CLI Tab Selection
> * ✅ Linode CLI Copy Code Snippet Icon
> * ✅ cURL Tab Selection
> * ✅ cURL Copy Code Snippet Icon
> * ✅ Integrations Tab Selection
> * ✅ Integrations - Ansible - Copy Code Snippet Icon
> * ✅ Integrations - Terraform - Copy Code Snippet Icon
> * ✅ SDKs Tab Selection
> * ✅ SDKs - Python - Copy Code Snippet Icon
> * ✅ SDKs - Go - Copy Code Snippet Icon

I will create a bug ticket to investigate why event emitters are not firing consistently for resources.

**Comment by cpathipa** [comment]
> > Metrics not firing for me:
> > 
> > * ❌ All Resource Links
> > 
> > Metrics working:
> > 
> > * ✅ Initial opening of modal
> > * ✅ Linode CLI Tab Selection
> > * ✅ Linode CLI Copy Code Snippet Icon
> > * ✅ cURL Tab Selection
> > * ✅ cURL Copy Code Snippet Icon
> > * ✅ Integrations Tab Selection
> > * ✅ Integrations - Ansible - Copy Code Snippet Icon
> > * ✅ Integrations - Terraform - Copy Code Snippet Icon
> > * ✅ SDKs Tab Selection
> > * ✅ SDKs - Python - Copy Code Snippet Icon
> > * ✅ SDKs - Go - Copy Code Snippet Icon
> 
> I will create a bug ticket to investigate why event emitters are not firing consistently for resources.

Looks like events are firing for the resources links 
image

**Review by hkhalil-akamai — APPROVED** [review]
Verified DX tools modal continues to work as expected and displays the button text based on the AB test value.

Verified LD events are being sent.

**Pull request merged**